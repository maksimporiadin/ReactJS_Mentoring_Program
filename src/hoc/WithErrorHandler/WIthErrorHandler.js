import React, { Component } from 'react';
import { Modal }  from '../../components/UI';
import axios from '../../axios';


const WithErrorHandler = (WrappedComponent) => {
    const WithErrorHandlerClass = class extends Component {
        state = {
            error: null,
        }

        componentWillMount () {
            this.reqInterceptor = axios.interceptors.request.use(req => {
                this.setState({ error: null });
                return req;
            });
            this.resInterceptor = axios.interceptors.response.use(res => res, err => {
                this.setState({ error: err });
            })
        }

        componentWillUnmount () {
            axios.interceptors.request.eject(this.reqInterceptor);
            axios.interceptors.response.eject(this.resInterceptor);
        }

        errorConfirmedHandler = () => {
            this.setState({ error: null });
        };

        render () {
            return (
                <React.Fragment>
                    <Modal
                        show={ this.state.error }
                        closeModal={ this.errorConfirmedHandler }>
                        { this.state.error ? this.state.error.message : null }
                    </Modal>
                    <WrappedComponent { ...this.props } />
                </React.Fragment>
            );
        }
    }

    return WithErrorHandlerClass;
}

export default WithErrorHandler;
