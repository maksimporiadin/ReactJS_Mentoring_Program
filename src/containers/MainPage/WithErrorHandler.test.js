import 'jsdom-global/register';
import React from 'react';
import WithErrorHandler from './index';
import { shallow } from 'enzyme';

describe('WithErrorHandler', () => {
    let wrapper;
    beforeEach(()=>{
        jest.resetAllMocks();
        wrapper = shallow(<WithErrorHandler />)
    });

    it('is renders', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('errorConfirmedHandler method test', function () {
        wrapper.setState({ error : true });

        wrapper.instance().errorConfirmedHandler();
        expect(wrapper.state('error')).toBe(null);
    });
});