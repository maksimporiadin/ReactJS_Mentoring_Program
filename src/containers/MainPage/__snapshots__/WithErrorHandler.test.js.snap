// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithErrorHandler is renders 1`] = `
<Fragment>
  <Modal
    closeModal={[Function]}
    show={null}
  />
  <Connect(MainPage) />
</Fragment>
`;
